docs/self-hosted/kubernetes-migration-guide.md

```diff
@@ -0,0 +1,144 @@
+
+```
+docker exec -it packagist-ui /srv/manager/bin/console packagist:self-hosted:migrate-storage export
+docker cp packagist-ui:/tmp/storage.tar.gz packagist_storage.tar.gz
+docker exec -it packagist-ui rm /tmp/storage.tar.gz
+docker stop packagist-ui
+```
+
+### Backup the PostgreSQL database
+
+At the end of this step, you should have a `packagist_db.sql` file in your working directory.
+
+```
+docker exec -it packagist-postgres pg_dump --clean --if-exists -U postgres -d packagist_db > packagist_db.sql
+docker stop packagist-postgres
+```
+
+### Backup the Redis database
+
+At the end of this step, you should have a `packagist_redis.rdb` file in your working directory.
+
+```
+docker exec -it packagist-redis /usr/local/bin/redis-backup.sh
+docker cp packagist-redis:/data/dump.rdb packagist_redis.rdb
+docker stop packagist-redis
+```
+
+### Start Private Packagist again (optional)
+
+Now that you have exported all the data, you may restart the Private Packagist application. This will allow your developers
+to continue with their work while you import the data in the Private Packagist Self-Hosted Kubernetes setup.
+
+```
+docker start packagist-postgres packagist-redis packagist-ui packagist-repo packagist-web packagist-worker    
+```
+
+Please note that any changes made in Private Packagist will now be lost and will have to be reapplied once everyone switches to Private Packagist Self-Hosted Kubernetes.
+
+## Installing Private Packagist Self-Hosted Kubernetes
+If you haven't done so already, now is the time to follow the [installation guide](./kubernetes.md) to install Private Packagist Self-Hosted Kubernetes.
+
+## Restore application data in Kubernetes
+
+All commands assume that you are logged in to the server where Private Packagist Self-Hosted Kubernetes is installed,
+your working directory contains the three files generated by the previous steps, and that
+Private Packagist Self-Hosted Kubernetes is installed in the Kubernetes namespace "default".
+
+To avoid any data inconsistency, we highly recommend going through the steps below in order and only restart the application
+once all the data has been restored.
+
+### Stop the Private Packagist application
+
+```
+kubectl scale deployment ui repo worker --replicas 0 -n default
+```
+
+### Restore the PostgreSQL database
+
+Follow the instructions below to restore the PostgreSQL database in the cluster.
+If you are using your own PostgreSQL instance, use the `packagist_db.sql` file directly to import the data into your instance.
+
+Please note that the backup will overwrite any existing data in the database and any changes you made during setup will be lost.
+
+```
+kubectl scale statefulset postgres --replicas 0 -n default
+kubectl delete pvc postgres-data-postgres-0 -n default
+kubectl scale statefulset postgres --replicas 1 -n default
+cat packagist_db.sql | kubectl exec -it -n default statefulsets/postgres -- psql -U postgres -d packagist_db
```

Comment:
Pretty tricky to review this now. Cause based on the conversation we had yesterday, we need to explain a lot about kubectl and/or how to verify that the commands worked?
Doesnt this need a bit time to scale up and down? in practice?